include/zephyr/dt-bindings/adc/silabs/xg24-adc.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2025 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This file was generated by the script gen_adc.py in the hal_silabs module.
+ * Do not manually edit.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_ADC_SILABS_XG24_ADC_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_ADC_SILABS_XG24_ADC_H_
+
+#define IADC_INPUT_GND      0
+#define IADC_INPUT_AVDD     16
+#define IADC_INPUT_IOVDD    17
+#define IADC_INPUT_DVDD     20
+#define IADC_INPUT_DECOUPLE 23
+#define IADC_INPUT_DAC0     32
+#define IADC_INPUT_DAC1     32
+#define IADC_INPUT_AIN0     64
+#define IADC_INPUT_AIN1     64
+#define IADC_INPUT_AIN2     80
+#define IADC_INPUT_AIN3     80
```

Comment:
Are these duplicated value expected?

Reply:
Yes, 0/2 is only for POS, 1/3 is only for NEG. This is a side effect of not having separate POS/NEG.